app/controllers/api/v1/ecf_participants_controller.rb

```diff
@@ -26,12 +27,8 @@ def index
 
     private
 
-      def serialized_response(profile)
-        relevant_induction_record = profile.relevant_induction_record(lead_provider:)
-
-        ParticipantFromInductionRecordSerializer
-          .new(relevant_induction_record)
-          .serializable_hash.to_json
+      def induction_records
+        @induction_records ||= ECFParticipants::Index.new(cpd_lead_provider: current_user, params:).induction_records
```

Comment:
Instead of passing in `cpd_lead_provider` might be better to do `lead_provider: current_user. lead_provider`, as thats what is actually needed in the service class.

Reply:
> Should the parameter here be cpd_lead_provider? if we're passing in current_user, its best to continue calling it that in the service class.

i think's it the other way. `current_user` is really not the best name and that should be changed to something more apt such as `current_cpd_lead_provider`

> Instead of passing in cpd_lead_provider might be better to do lead_provider: current_user. lead_provider, as thats what is actually needed in the service class.

I guess this is preferential. should the controller/higher level class care about the details of what gets passed around? do we want future classes like this one to have different interfaces eg all accept accepting `CpdLeadProvider` vs some take `LeadProvider` whilst others take `NPQLeadProvider` and the controller/developer needs to figure out what to pass in?

Reply:
The service class expects LeadProvider and it makes sense to pass it as a parameter.
But I guess overall it's very confusing to pass in `current_user`, which is a `User` object, but the name suggests a `CpdLeadProvider`.